influxdb_client/client/bucket_api.py

```diff
@@ -56,7 +55,48 @@ def create_bucket(self, bucket=None, bucket_name=None, org_id=None, retention_ru
                                                                   client=self._influxdb_client,
                                                                   required_id=True))
 
-        return self._buckets_service.post_buckets(post_bucket_request=bucket)
+        try:
+            return self._buckets_service.post_buckets(post_bucket_request=bucket)
+        except ApiException:
+            # Fall back to v1 API if buckets are not supported
+            database_name = bucket_name if bucket_name is not None else bucket
```

Comment:
This approach is too general and can lead to user confusion. The better way would be to create a separate API for InfluxDB 1.8 something like `influxdb_18_api.py`.

Reply:
Thanks for reviewing.

I would disagree to create a separate API (If I understand your suggestion right). The goal should be that code changes are not necessary regardless of what version of InfluxDB a user uses (1.8 or 2.0 and above).

`create_bucket()` should always succeed for both versions without extra error handling. Otherwise users could continue using the old influxdb-client for 1.8 and this one for >2.0 .

My suggestion would be to implement a more specific exception handler. I agree that just `ApiException` is too general. I will have a look into the returned `ApiException.code`.

Would this be acceptable for your project?

Reply:
We can accept this if the behaviour will be described in the documentation. Instead of catching exception you can check the InfluxDB version, something like here: https://github.com/influxdata/influxdb-client-python/blob/f3346d95e81e56186aee93ca11f068f8c415c62e/influxdb_client/service/_base_service.py#L28